state.go

```diff
@@ -616,31 +616,34 @@ func (m *Memberlist) gossip() {
 		bytesAvail -= encryptOverhead(m.encryptionVersion())
 	}
 
-	for _, node := range kNodes {
-		// Get any pending broadcasts
-		msgs := m.getBroadcasts(compoundOverhead, bytesAvail)
-		if len(msgs) == 0 {
-			return
+	msgs := m.getBroadcasts(compoundOverhead, bytesAvail, len(kNodes))
```

Comment:
In this file:
- move get broadcasts outside of loop
- build messages once and ensure the same messages (and improved processing) gets sent to each node.